react-components/src/query/network/searchClassicAssetsForModels.ts

```diff
@@ -0,0 +1,70 @@
+import { type Asset, type CogniteClient } from '@cognite/sdk';
+
+import { type TaggedAddResourceOptions } from '../../components/Reveal3DResources/types';
+import { type RevealRenderTarget } from '../../architecture';
+import { searchClassicAssetsForCadModels } from './searchClassicCadAssets';
+import { searchClassicPointCloudAssets } from './searchClassicPointCloudAssets';
+import { isClassicIdentifier } from '../../components/Reveal3DResources/typeGuards';
+import { uniqBy } from 'lodash';
+
+export type SearchClassicAssetsResponse = {
+  nextCursor: string | undefined;
+  data: Asset[];
+};
+
+export async function searchClassicAssetsForModels(
+  searchQuery: string,
+  resources: TaggedAddResourceOptions[],
+  limit: number,
+  cadAssetsCursor: string | undefined,
+  sdk: CogniteClient,
+  renderTarget: RevealRenderTarget
+): Promise<SearchClassicAssetsResponse> {
+  const isFirstPage = cadAssetsCursor === undefined;
+
+  const assetMappingAndNode3DCache = renderTarget.cdfCaches.assetMappingAndNode3dCache;
+
+  const cadModels = resources
+    .filter((resource) => resource.type === 'cad')
+    .map((resource) => resource.addOptions)
+    .filter(isClassicIdentifier);
+
+  const pointClouds = resources
+    .filter((resource) => resource.type === 'pointcloud')
+    .map((resource) => resource.addOptions)
+    .filter(isClassicIdentifier);
+
+  /* const image360Collections = resources
```

Comment:
Please remove commented out code or comment above that this will be used in next PR